### DISCUSSION TITLE
Integration of non-PyPi/conda-managed codes

### DISCUSSION THREAD

**Discussion by matiscke** [rodluger/showyourwork#346] [Q&A] [answered]
I'm writing an article whose results and figures rely on a github-hosted python tool that is currently not managed via PyPi or conda. What is the best way to integrate it in my showyourwork project?

**Comment by dfm** [comment, marked as answer]
@matiscke — My usual approach to this would be to install the tool using pip (which can directly install from GitHub) by adding it to your `environment.yml` file:

```yaml
dependencies:
  - ...
  - pip
  - pip:
      - git+https://github.com/showyourwork/showyourwork.git@main
```

If you can't use pip to install the tool, we could probably hack something with snakemake by adding custom rules to your `Snakefile`. This might be something like (note: I haven't actually tried this...):

```python
rule install_tool:
    output:
        "installed.flag"
    conda:
        "environment.yml"
    shell:
        # ... install your tool using bash commands
        touch installed.flag
```

And then add `installed.flag` as an input dependency to your scripts. Hope this helps!

**Reply by matiscke** [reply]
Your first solution works, thanks a lot @dfm !!
